### ISSUE TITLE
Ent Web UI Authorization stopped working after bump from 6.1.3 to 6.2.0

### ISSUE THREAD

**Issue by raivil** [mperham/sidekiq#4843]
Ruby version: 2.7.0
Rails version: 6.1.3
Sidekiq / Pro / Enterprise version(s): `6.2.0`/`5.2.1`/`2.2.2`

The application is just a simple Rails App with devise, omniauth, and SidekiqWebUI. 
After upgrading from sidekiq 6.1.3 to 6.2.0, authorization (`Sidekiq::Web.authorize`) is no longer working. The block is not called at all.
Solution is pretty much what's describe at https://github.com/mperham/sidekiq/wiki/Ent-Web-UI.

Code snippets:
```
Sidekiq.configure_client do |config|
  config.redis = { url: Redis.redis_url(db_key), driver: "hiredis" }
end
```

```
# frozen_string_literal: true

require "sidekiq-ent/web"

Sidekiq::Web.authorize do |env, method, _path|
  session = env["rack.session"]
  warden_key = session["warden.user.user.key"]
  return false unless warden_key && warden_key[0] && warden_key[0][0]

  user = User.find(warden_key[0][0])
  (method == "GET" && user.readonly_user?) || user.valid_user?
end

Rails.application.routes.draw do
  devise_for :users, controllers: { omniauth_callbacks: "users/omniauth_callbacks" }

  root to: "main#index"

  authenticated :user, ->(u) { u.valid_user? } do
    mount Sidekiq::Pro::Web.with(redis_pool: REDIS_POOL), at: "/sidekiq", as: "sidekiq"
  end
end
```

**Comment by mperham** [comment]
It seems to work for my app.  Can you try this mount and see if it makes a difference?

```ruby
mount Sidekiq::Web, at: "/sidekiq"
```

**Comment by raivil** [comment]
yes, it makes a difference. 
That version will work and requests will be blocked.

**Comment by mperham** [comment]
Ok, sounds like there was a slight semantic change but it's fixed for you now.

**Comment by raivil** [comment]
ok, but how can I specify the `redis_pool` if not using `Sidekiq::Pro::Web`?

**Comment by mperham** [comment]
Why do you need to specify anything? Sidekiq::Pro::Web is built into Sidekiq::Web. My `mount` and your `mount` are functionally equal, no?

**Comment by raivil** [comment]
Planning on adding multiple sidekiq web dashboards/shards. (https://github.com/mperham/sidekiq/wiki/Pro-Web-UI#sharding).
Is there another way to do that? Am I missing anything?

**Comment by mperham** [comment]
Gotcha, you're trying to use Authorization and Sharding at once. Unfortunately sharding creates a subclass of Sidekiq::Web which does not inherit Sidekiq::Web config. This will work:

```ruby
  # in config/routes.rb
  shard = Sidekiq::Pro::Web.with(redis_pool: Sidekiq.redis_pool)
  shard.use Sidekiq::Enterprise::Web::Authorization do |env, method, path|
    path != "/nope" # FIXME
  end
  mount shard, at: '/sidekiq', as: 'sidekiq'
```

You can create as many shards as you want.

**Comment by mperham** [comment]
I've fixed the issue on Sidekiq Pro master and it will be included in 5.3.0 so this will work in the future:

```ruby
  Sidekiq::Web.authorize do |env,method,path|
    path != "/nope"
  end
  mount Sidekiq::Pro::Web.with(redis_pool: Sidekiq.redis_pool), at: '/sidekiq', as: 'sidekiq'
```
